### PR TITLE
improve: condense computeSystem fee

### PR DESCRIPTION
**Pull request by james-a-morris**

### CODE REVIEW

File: src/clients/UBAClient/UBAClientAbstract.ts [hunk 1]
```diff
@@ -142,23 +142,21 @@ export abstract class BaseUBAClient {
...
-   * @param depositChain The chainId of the deposit
-   * @param refundChain The chainId of the refund
+   * @param destinationChainId The chainId of the transaction
    * @param spokePoolToken The token to get the system fee for
    * @param amount The amount to get the system fee for
    * @param hubPoolBlockNumber The block number to get the system fee for
    * @returns The system fee for the given token on the given chainId at the given block number
    */
   public async computeSystemFee(
-    depositChain: number,
```
Comment (pxrl) [inline src/clients/UBAClient/UBAClientAbstract.ts:153]: Don't we still need the depositChainId? I'd thought that would be retained, but I might have misunderstood.

Comment (james-a-morris) [reply src/clients/UBAClient/UBAClientAbstract.ts:153]: I was of the understanding that this was different. Fair point though, I've updated it

File: src/clients/UBAClient/UBAClientAbstract.ts [hunk 2]
```diff
@@ -142,23 +142,23 @@ export abstract class BaseUBAClient {
...
+    destinationChainId: number,
     spokePoolToken: string,
     amount: BigNumber,
     hubPoolBlockNumber: number
   ): Promise<SystemFeeResult> {
     const [lpFee, { balancingFee: depositBalancingFee }] = await Promise.all([
-      this.computeLpFee(spokePoolToken, depositChain, refundChain, amount),
-      this.computeBalancingFee(spokePoolToken, amount, hubPoolBlockNumber, depositChain, UBAActionType.Deposit),
+      this.computeLpFee(spokePoolToken, depositChainId, destinationChainId, amount),
+      this.computeBalancingFee(spokePoolToken, amount, hubPoolBlockNumber, destinationChainId, UBAActionType.Deposit),
```
Comment (nicholaspai) [inline src/clients/UBAClient/UBAClientAbstract.ts:161]: Why did you change this line? I thought this balancing fee would be based on the deposit chain only?

File: src/clients/UBAClient/UBAClientAbstract.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientAbstract.ts:153]: I also agree we need deposit chain ID

System Fee = deposit balancing fee + LP fee
- deposit balancing fee is based only on deposit chain ID
- LP fee is based on deposit chain and destination chain

**Review by nicholaspai — COMMENTED** [review]
I want to make sure we're aligned https://github.com/across-protocol/sdk-v2/pull/248/files#r1235906195

File: src/clients/UBAClient/UBAClientAbstract.ts (hunk 2 above)
Comment (james-a-morris) [reply src/clients/UBAClient/UBAClientAbstract.ts]: This was a reversion from testing. Resolved here: https://github.com/across-protocol/sdk-v2/pull/248/commits/ee8bab88bf45089a1fbaa2d03fb84c11c59e922a

**Review by nicholaspai — APPROVED** [review]
Thanks for responding to comments LGTM

**Review by pxrl — APPROVED** [review]
Also LGTM
